### ISSUE TITLE
PublishTweet no exception for "Invalid or expired token"

### ISSUE THREAD

**Issue by daikoz** [linvi/tweetinvi#768]
Hi,

Follow this code:

```C#
ITwitterCredentials credentials = Auth.SetUserCredentials(website.TwitterConsumerKey, website.TwitterConsumerSecret, authorizationKey, authorizationSecret);
ITweet theTweet = await TweetAsync.PublishTweet(message);
ITwitterException ex = ExceptionHandler.GetLastException();
```

With wrong credential (user has remove its authorization), PublishTweet return null.
**BUG:** ExceptionHandler.GetLastException(); return no exception !!!

But if I don't use Async:

```C#
ITwitterCredentials credentials = Auth.SetUserCredentials(website.TwitterConsumerKey, website.TwitterConsumerSecret, authorizationKey, authorizationSecret);
ITweet theTweet = Tweet.PublishTweet(message);
ITwitterException ex = ExceptionHandler.GetLastException();
```

With wrong credential (user has remove its authorization), PublishTweet return null.
And ex contain ""Unauthorized -  Authentication credentials were missing or incorrect.""
The behavior is OK.

Regards

**Comment by JoshKeegan** [comment]
Hi,
Thanks for the bug report & repro code.
I've submitted a PR fixing this: #770.

Here's a build containing the fix if you want to try it out:
[netstandard1.4.zip](https://github.com/linvi/tweetinvi/files/2520671/netstandard1.4.zip)

**Comment by zokiad** [comment]
Hi,
ok, fix work. I have now exception.

**But, there are an other bug with:**

ExceptionHandler.SwallowWebExceptions = false;

With async, no exception is throw (but ExceptionHandler.GetLastException() work).
Without async, exception is throw and ExceptionHandler.GetLastException() work.

**Comment by JoshKeegan** [comment]
Thanks! I'll look at that.

**Comment by zokiad** [comment]
I tested our new build. I think there are an issue with this build because I have all bugs (first case and second case).

**Comment by JoshKeegan** [comment]
Can you try this new build?
[netstandard2.0.zip](https://github.com/linvi/tweetinvi/files/2719553/netstandard2.0.zip)

If you're using `Task.Run` or other ways of running things on other threads for managing multiple Twitter user's credentials, please read through the "Usage" section of #770.

**Comment by zokiad** [comment]
ok work fine

**Comment by linvi** [comment]
Thanks for the feedback.

**Comment by linvi** [comment]
This issue will be fixed by Tweetinvi 5.0 as it no longer relies on `ThreadStatic` or `AsyncContext`.
